src/transformers/generation/utils.py

```diff
@@ -888,6 +895,29 @@ def _reorder_cache(self, past_key_values, beam_idx):
             f" enable beam search for {self.__class__}"
         )
 
+    def _get_candidate_generator(
```

Comment:
This function will be expanded as we add more `CandidateGenerator` :)

Reply:
@apoorvumang exactly!

Reply:
got it, so I'll write a `PromptLookupCandidateGenerator` that implements `CandidateGenerator`, and then wire it up in this function

Reply:
@ArthurZucker precisely, we will have flags to control which candidate generation strategies we have in place. I suspect that, because some candidate generation strategies are so cheap (like the one proposed in #27722), assisted generation may become mainstream!